c2corg_api/views/feed.py

```diff
@@ -182,24 +191,46 @@ def create_personal_filter(user):
     """ Create a filter condition for the query to get the changes taking
     the filter preferences of the user into account.
     """
+
+    # if user.feed_filter_lang_preferences:
+
     if user.feed_followed_only:
         # only include changes of followed users
         return create_followed_users_filter(user)
 
     # filter on area and activity (`and` connected)
-    area_activity_filter = None
-    if user.feed_filter_activities or user.has_area_filter:
+    area_activity_language_filter = None
+
+    if user.feed_filter_activities or user.has_area_filter or \
+            user.feed_filter_lang_preferences is not None:
         area_filter = create_area_filter(user)
         activity_filter = create_activity_filter(user)
+        language_filter = create_language_filter(user)
+
+        if area_filter is not None and \
+                activity_filter is not None and \
+                language_filter is not None:
+            area_activity_language_filter = and_(
+                area_filter, activity_filter, language_filter)
+
+        elif area_filter is not None and activity_filter is not None:
+            area_activity_language_filter = and_(
+                area_filter, activity_filter)
+        elif activity_filter is not None and language_filter is not None:
+            area_activity_language_filter = and_(
+                activity_filter, language_filter)
+        elif language_filter is not None and area_filter is not None:
+            area_activity_language_filter = and_(
+                language_filter, area_filter)
 
-        if area_filter is not None and activity_filter is not None:
-            area_activity_filter = and_(area_filter, activity_filter)
         elif area_filter is not None:
-            area_activity_filter = area_filter
+            area_activity_language_filter = area_filter
         elif activity_filter is not None:
-            area_activity_filter = activity_filter
+            area_activity_language_filter = activity_filter
+        elif language_filter is not None:
+            area_activity_language_filter = language_filter
```

Comment:
All those conditions seem a bit complicated.
Why don't you use the following approach:
```
filters = []
if area_filter is not None:
    filters.append(area_filter)
if activity_filter is not None:
    filters.append(activity_filter)
if language_filter is not None:
    filters.append(language_filter)
if len(filters):
    area_activity_language_filter = and_(filters)
```
Perhaps you don't even need the ``is not None`` explicit conditions?